images/pdc-update/src/cost/age.rs

```diff
@@ -0,0 +1,49 @@
+use crate::metrics::MetricsValue;
+
+use super::CostModel;
+
+/// [`CostModel`] using the age of the processing
+#[derive(Debug, Default, Clone, PartialEq, PartialOrd)]
+pub struct AgeCost {
+    /// Threshold below which the actual age is ignored
+    age_threshold: f64,
+    /// Scaling factor for the cost before conversion to i32
+    scaling: f64,
+}
+
+impl AgeCost {
+    /// Build a new [`AgeCost`]
+    pub fn new(age_threshold: f64, granularity: f64) -> Self {
+        Self {
+            age_threshold,
+            scaling: 1.0 / granularity,
+        }
+    }
+}
+
+impl CostModel for AgeCost {
+    fn metrics_to_cost(&self, metrics: &MetricsValue) -> i32 {
+        let duration = metrics.timestamp - metrics.value;
+        let cost = if duration <= 0.0 {
+            -1.0
+        } else if duration <= self.age_threshold {
+            self.age_threshold
+        } else {
+            f64::from(i32::MAX) - metrics.value
+        };
+        let cost = cost * self.scaling;
+
+        let cost = cost.round().min(f64::from(i32::MAX)).max(-1.0) as i32;
+
+        log::trace!(
+            "Compute cost for {}/{} = {} @{} -> {}",
```

Comment:
why are you using two differents syntax one here and one in ling 60 in pod.rs? Is it strictly similar?

Reply:
You can write the name directly between braces, but it works only for plain names, not for expressions.
Here, I have `metrics.*` so it will not accept it in the braces.